aws/ec2.go

```diff
@@ -0,0 +1,225 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	awssession "github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	defaultTimeOut = 20
+	defaultRetries = 3
+)
+
+func convertError(err error) string {
+	if awsErr, ok := err.(awserr.Error); ok {
+		return fmt.Sprintf("%s: %s", awsErr.Code(), awsErr.Message())
+	}
+
+	return fmt.Sprintf("%v", err.Error())
+}
+
+func retriable(err error) bool {
+	if awsErr, ok := err.(awserr.Error); ok {
+		switch awsErr.Code() {
+		case "InternalError":
+			return true
+		case "InternalFailure":
+			return true
+		case "RequestLimitExceeded":
+			return true
+		case "ServiceUnavailable":
+			return true
+		case "Unavailable":
+			return true
+		}
+	}
+
+	return false
+}
+
+func checkSourceDestinationValueIsDisable(check string) bool {
```

Comment:
Do we really need a function just to do a string comparison? Why not just do the comparison inline?

Reply:
For UT.

Reply:
I don't think it's worth it. We don't need a UT for the `==` operator.

Reply:
It's more about checking the valid values. Please have a look at the testing section and the idea that by default the input to set-src-check-dst will always be true unless set to false by `disable`.

Reply:
Agree with Casey, I think it'd be clearer to check inline.